src/reaper_osara.cpp

```diff
@@ -2261,6 +2285,83 @@ bool handleMainCommandFallback(int command, int flag) {
 	return false;
 }
 
+/*** A control surface to obtain certain info that can only be retrieved that way.
+ */
+class Surface : IReaperControlSurface {
```

Comment:
I think this should be moved into a separate source file; e.g. surface.cpp. There's not much code here now, but I imagine this will only get larger over time, and reaper_osara.cpp is already pretty large as it is. This will mean you'll need to export some functions and variables from reaper_osara.cpp in osara.h.

Rather than having to declare the whole class in surface.h (which is just annoying), I'd probably create registerSurface/unregisterSurface functions which are exported and get called from reaper_osara.cpp.